File: app/src/main/java/com/ikvych/cocktail/data/repository/base/DrinkDbRepository.java
package com.ikvych.cocktail.data.repository.base;

import androidx.lifecycle.LiveData;

import com.ikvych.cocktail.data.entity.Drink;

import java.util.List;

public interface DrinkDbRepository {

    LiveData<List<Drink>> getDrinks();

    void saveDrink(Drink drink);
}

File: app/src/main/java/com/ikvych/cocktail/data/repository/base/DrinkDbRepository.kt
package com.ikvych.cocktail.data.repository.base

import androidx.lifecycle.LiveData
import com.ikvych.cocktail.data.entity.Drink


interface DrinkDbRepository {

    fun getDrinks(): LiveData<List<Drink>>

    fun getFavoriteDrinks(): LiveData<List<Drink>>

    fun saveDrink(drink: Drink)

    fun removeDrink(drink: Drink)

    fun findDrinkById(drinkId: Long): Drink

    fun findDrinkByName(drinkName: String): Drink
}